packages/components/src/styles/components/form/layout.scss

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// FORM LAYOUT COMPONENTS
+//
+
+// FORM
+
+// Default max-width for form sections & similar components
+$hds-form-section-max-width: 768px;
+
+.hds-form {
+  --hds-form-section-max-width: #{$hds-form-section-max-width};
+
+  display: flex;
+  flex-direction: column;
+  gap: 32px;
+
+  // Set a max-width for the ButtonSet when it's within an Hds::Form
+  // Q: Should this be done (max-width on ButtonSet) and if so should it be done here or in the ButtonSet component?
+  .hds-button-set {
+    max-width: var(--hds-form-section-max-width, #{$hds-form-section-max-width});
+  }
+}
+
+// FORM HEADER & SECTION HEADER
+
+.hds-form__header,
+.hds-form__section-header {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+}
+
+// Note: Only the Form Header needs a max-width as the Section Header is used within a Section
+.hds-form__header {
+  max-width: var(--hds-form-section-max-width, #{$hds-form-section-max-width});
+}
+
+// hasMaxWidth = false
+.hds-form__header--has-max-width-false {
+  --hds-form-section-max-width: none;
+}
+
+// FORM SECTION
+
+.hds-form__section {
+  display: flex;
+  flex-direction: column;
+  gap: 24px;
+  max-width: var(--hds-form-section-max-width, #{$hds-form-section-max-width});
+}
+
+// hasMaxWidth = false
+.hds-form__section--has-max-width-false {
+  --hds-form-section-max-width: none;
+}
+
+// FORM DIVIDER
+
+.hds-form__divider {
+  max-width: var(--hds-form-section-max-width, #{$hds-form-section-max-width});
+  height: 0;
+  border-top: 1px solid var(--token-color-border-primary);
+}
+
+// hasMaxWidth = false
+.hds-form__divider--has-max-width-false {
```

Comment:
[thought] what if we have a common, more generic (but still form-related) CSS class that can be used for all the elements, instead of defining one for each one of them?

Reply:
Oh, I updated all the classes earlier to get rid of the "false" naming. I guess I could use a generic class for them all instead. Let me see...

Reply:
I don't know a good name for this general class.

Maybe `.hds-form-content--has-max-width`? I can use that for now anyway...

Reply:
Or maybe `hds-form-constrain-width`?

Reply:
Went ahead and updated this. Can change the classname later if you have better ideas.